megamek/src/megamek/common/logging/DefaultMmLogger.java

```diff
@@ -368,36 +455,76 @@ public void warning(final Object callingObject, final String message) {
                                            final T throwable) {
         return log(callingObject.getClass(), getCallingMethod(), LogLevel.WARNING, message, throwable);
     }
+    
+    @Override
+    public <T extends Throwable> T warning(final String message, final T throwable) {
+        return log(getCallingClass(), getCallingMethod(), LogLevel.WARNING, message, throwable);
+    }
     //endregion Warning
     
     @Override
     public void methodBegin(final Class<?> callingClass, final String methodName) {
         log(callingClass, methodName, LogLevel.DEBUG, METHOD_BEGIN + methodName);
     }
     
+    @Override
+    public void methodBegin() {
+        log(getCallingClass(), getCallingMethod(), LogLevel.DEBUG, METHOD_BEGIN, null);
+    }
+
+    
     @Override
     public void methodEnd(final Class<?> callingClass, final String methodName) {
         log(callingClass, methodName, LogLevel.DEBUG, METHOD_END + methodName);
     }
+    
+    @Override
+    public void methodEnd() {
+        log(getCallingClass(), getCallingMethod(), LogLevel.DEBUG, METHOD_END, null);
+    }
 
     @Override
     public void methodCalled(final Class<?> callingClass, final String methodName) {
         log(callingClass, methodName, LogLevel.DEBUG, METHOD_CALLED + methodName);
     }
+    
+    @Override
+    public void methodCalled() {
+        log(getCallingClass(), getCallingMethod(), LogLevel.DEBUG, METHOD_CALLED, null);
+    }
+
 
     /**
-     * Retrieves the name of the method calling log/info/error etc.
-     * from the stack trace.
+     * Retrieves the name of the method and the line number 
+     * calling log/info/error etc. from the stack trace.
      *
      * This method MUST be called from a top-level method (such as info or fatal)
      * as it depends on the call stack depth of exactly three methods between
      * internally calling getStackTrace and the user's code.
      */
     private String getCallingMethod() {
         try {
-            return Thread.currentThread().getStackTrace()[3].getMethodName() + "()";
+            String result = Thread.currentThread().getStackTrace()[3].getMethodName() + "()";
+            result += ", line " + Thread.currentThread().getStackTrace()[3].getLineNumber();
+            return result;
+        } catch (Exception e) {
+            return "DefaultMMLogger Error: Could not obtain method name.";
+        }
+    }
+    
+    /**
+     * Retrieves the name of the class calling log/info/error etc.
+     * from the stack trace.
+     *
+     * This method MUST be called from a top-level method (such as info or fatal)
+     * as it depends on the call stack depth of exactly three methods between
+     * internally calling getStackTrace and the user's code.
+     */
+    private String getCallingClass() {
+        try {
+            return Thread.currentThread().getStackTrace()[3].getClassName();
```

Comment:
Does dynamically retrieving this information e.g. class name, method name have any negative performance impact? Probably not, compared to disk I/O, but worth asking.

Reply:
It shouldn't, but.. I'll test on my potato laptop before we merge the story just to be on the safe side